zenodo/modules/utils/cli.py

```diff
@@ -407,3 +409,32 @@ def update_openaire_communities(path):
         unresolved_communities, indent=4, separators=(', ', ': '))))
     click.secho('{0}'.format(json.dumps(mapping, indent=4,
                                         separators=(', ', ': '))), fg='blue')
+
+
+@utils.command('list_failed_openaire_records')
+@with_appcontext
+def list_failed_openaire_records():
+    """Lists records that failed to get indexed in OpenAIRE."""
+    failed_record_recids = []
+    for key in current_cache.cache._read_clients.scan_iter(
+        match=(current_cache.cache.key_prefix + 'openaire_direct_index*')):
```

Comment:
This iteration logic can be put in an `_iter_openaire_direct_index_keys()` helper function.